Assets/Scripts/Enemy/BossAbility.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using ICSharpCode.NRefactory.Ast;
+using Photon.Pun;
+using UnityEngine;
+using UnityEngine.Experimental.Rendering.Universal;
+
+namespace Enemy
+{
+    [Serializable] public abstract class BossAbility : MonoBehaviourPun
+    {
+        [SerializeField] private float castTime;
+        [SerializeField] private bool immobilizeWhilePerforming;
+        //public Animation moveAnimation;
+        
+        protected Light2D _light2D;
+        protected ChaserAI _chaserAI;
+        private bool _duringPerformAction;
+        protected GameObject referenceObject;
+
+        private void Start()
+        {
+            referenceObject = transform.parent.gameObject;
+            _light2D = GetComponentInParent<Light2D>();
+            _chaserAI = GetComponentInParent<ChaserAI>();
+        }
+
+        private void Update()
+        {
+            if (_duringPerformAction) DuringPerformActionClient();
+        }
+
+        /// <summary>
+        /// Every time a move is used, this is called
+        /// </summary>
+        /// <param name="bossMove"></param>
+        /// <param name="routine"></param>
+        /// <param name="castTime"></param>
+        /// <param name="immobilizeWhilePerforming"></param>
+        /// <returns></returns>
+        // Gets called by the master client only
+        public IEnumerator PerformActionMaster()
+        {
+            if (immobilizeWhilePerforming) _chaserAI.DisableMovement(true);
+            
+            yield return new WaitForSeconds(castTime);
```

Comment:
It might complicate a few things as the boss movement is paused until the castTime is up, are you suggesting that the `_chaserAI.DisableMovement(false);` line also be put in AbilityCoroutine then?